### PR TITLE
[reconfigurator] `BlueprintBuilder` cleanup 1/5 - move `BlueprintDisksEditor` to its own module

### PR DESCRIPTION
**Pull request by jgallagher**
This is _mostly_ moving code around; specifically, `BlueprintDisksEditor` is now in its own module. The non-"moving code around" changes are:

* The API stile is more imperative.
* Generation 1 for `OmicronPhysicalDisksConfig` is now only the empty set of disks; if there are disks present, that will be at least generation 2. This caused some expectorate churn but seems probably fine? It matches how `OmicronZonesConfig` generations are numbered.

### CODE REVIEW

File: nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    #[cfg(test)]
+    pub fn current_sled_disks(
+        &self,
+        sled_id: &SledUuid,
+    ) -> Option<&BTreeMap<PhysicalDiskUuid, BlueprintPhysicalDiskConfig>> {
+        let config = self.current.get(sled_id)?;
+        Some(&config.disks)
+    }
+
+    pub fn build(
```
Comment (smklein) [inline nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs:62]: Could we include some docs about why we're passing `sled_ids` again here?

(e.g., why are we passing this here, when `BlueprintDisksEditor` is already taking a map of sleds as input?)

File: nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+#[derive(Debug)]
+pub(super) struct SledDisksEditor<'a> {
+    config: &'a mut DisksConfig,
+    changed: bool,
+    sled_id: SledUuid,
+    parent_changed_set: &'a mut BTreeSet<SledUuid>,
+}
+
+impl Drop for SledDisksEditor<'_> {
```
Comment (smklein) [inline nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs:93]: Neat, so the mutable borrows here require that `SledDisksEditor` is dropped before `BlueprintDisksEditor::build` gets called, right?

**Review by smklein — APPROVED** [review]

File: nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs (hunk 1 above)
Comment (andrewjstone) [reply nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs:62]: I agree with Sean that a comment is a good idea.

 To check my understanding, the sleds we take in the constructor are all from the parent blueprint, but these sleds are the `InService` sleds from the `PlanningInput` which may be different. This allows us to add a `DiskConfig::Empty` for any new sleds, and also to ignore sleds that are no longer in service but may have configurations.

**Review by andrewjstone — APPROVED** [review]
Looks great.

File: nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs (hunk 2 above)
Comment (jgallagher) [reply nexus/reconfigurator/planning/src/blueprint_builder/builder/disks_editor.rs:93]: Correct.

You could always leak or `mem::forget()` a `SledDisksEditor`, but at that point you get what you deserve, I think? The alternative would be for `sled_disks_editor` to take an `FnOnce(&mut SledDisksEditor)`, but that's much more awkward from a caller's point of view. It seems fine to say "if you leak a `SledDisksEditor` your edits won't be fully reflected".

**Pull request merged**